src/juju/reducers.js

```diff
@@ -0,0 +1,24 @@
+import produce from "immer";
+
+import { actionsList } from "./actions";
+
+/* eslint-disable default-case */
+// immer handles the default case so one isn't needed.
+export default function jujuReducers(state = {}, action) {
```

Comment:
Does this seem like a good architecture for this? I'll add the test if this approach is 👍

Reply:
I think my only advice around modeling state in redux is to try to keep state as flat and close to the API as possible, and do all your transformations/clever stuff in selectors.

Reply:
I suppose it would depend on how you'll be accessing that data. If you're always going to be accessing it in a specific format, you might as well store it in that format instead of re-creating that format on every request.

In this case we will be presenting this model list data to the tables to render in its entirety. I feel that a selector would be used if it was a subset of data or not the primary use case for it.